src/lib/libast/features/mode.c

```diff
@@ -27,192 +27,22 @@
  * generate mode features
  */
 
-#include "limits.h"
-
-#include "FEATURE/param"
-
-#include <modecanon.h>
+#include <stdio.h>
+#include <fcntl.h>
 
 int
 main()
 {
-	int	n;
-	int	idperm;
-	int	idtype;
-
-	idperm = idtype = 1;
 #ifndef S_ITYPE
 #ifdef	S_IFMT
 	printf("#define S_ITYPE(m)	((m)&S_IFMT)\n");
 #else
 	printf("#define S_ITYPE(m)	((m)&~S_IPERM)\n");
```

Comment:
Ugh! I've never heard of the `S_IPERM` or `S_ITYPE` symbol. It's not defined on macOS or Linux. I can't find them mentioned in the APUE book either. So either this feature test can be completely eliminated or there are platforms we care about which have this unusual configuration. If the latter a comment to that effect would be a good idea.

Reply:
None of these macros are provided by default and there is some ast code that depend on them. I plan to clean it up later.

Reply:
SGTM, ship it.

Reply:
Ehmm, if it is not ready, it should never be shipped/commited! Wondering, where you learned to do such insane things ...

Reply:
@jelmd, Your comments are not constructive and their tone is condescending. AFAICT you haven't actually contributed any changes. On the contrary, you've made it clear you won't be contributing. In issue #40 I suggested that you or @marcastel could contribute by helping port some of the Solaris patches to ksh93 to this project. You snidely replied:

> It was just a hint, to possibly safe you some time. If you prefer troubleshooting it by your own and provide your own fix, just go ahead.

So the next time you start to write a comment might think about what I've said if you want anyone to pay attention to what you have to say.